docs/develop/go/message-passing.mdx

```diff
@@ -27,135 +27,178 @@ tags:
+			}
+			return nil
+		},
+	})
+  ...
+}
+```
 
-### Send a Signal from a Temporal Client {#send-signal-from-client}
+- Register an Update handler for a given name using either [workflow.SetUpdateHandler](https://pkg.go.dev/go.temporal.io/sdk/workflow#SetUpdateHandler) or [workflow.SetUpdateHandler](https://pkg.go.dev/go.temporal.io/sdk/workflow#SetUpdateHandlerWithOptions).
+- You must include a `workflow.Context` parameter in the first position of the function.
+- The function can return either a serializable value with an error or just an error.
+- To reject an update you must return an error or panic in the validator.
+- Validators are optional; if you don't want to be able to reject updates then you don't need a validator.
+- The validator must take the same argument type as the handler, but always returns an error.
+- The Workflow's WorkflowPanicPolicy determines how panics are handled inside the Handler function.
 
-**How to send a Signal from a Temporal Client using the Go SDK.**
+## Sending Messages
 
-When a Signal is sent successfully from the Temporal Client, the [WorkflowExecutionSignaled](/references/events#workflowexecutionsignaled) Event appears in the Event History of the Workflow that receives the Signal.
+### Sending a Query {#send-query}
 
-Use the `SignalWorkflow()` method on an instance of the [Go SDK Temporal Client](https://pkg.go.dev/go.temporal.io/sdk/client#Client) to send a [Signal](/encyclopedia/workflow-message-passing#sending-signals) to a [Workflow Execution](/workflows#workflow-execution).
+Queries are sent from a Temporal Client.
 
-Pass in both the [Workflow Id](/workflows#workflow-id) and [Run Id](/workflows#run-id) to uniquely identify the Workflow Execution.
-If only the Workflow Id is supplied (provide an empty string as the Run Id param), the Workflow Execution that is Running receives the Signal.
+Use [`Client.QueryWorkflow`](https://pkg.go.dev/go.temporal.io/sdk/client#Client.QueryWorkflow) or [`Client.QueryWorkflowWithOptions`](https://pkg.go.dev/go.temporal.io/sdk/client#Client.QueryWorkflowWithOptions).
 
 ```go
 // ...
-signal := MySignal {
-  Message: "Some important data",
+supportedLangResult, err := temporalClient.QueryWorkflow(context.Background(), we.GetID(), we.GetRunID(), message.GetLanguagesQuery, message.GetLanguagesInput{IncludeUnsupported: false})
+if err != nil {
+    log.Fatalf("Unable to query workflow: %v", err)
 }
-err = temporalClient.SignalWorkflow(context.Background(), "your-workflow-id", runID, "your-signal-name", signal)
+var supportedLang []message.Language
+err = supportedLangResult.Get(&supportedLang)
 if err != nil {
-	log.Fatalln("Error sending the Signal", err)
-	return
+    log.Fatalf("Unable to get query result: %v", err)
 }
+log.Println("Supported languages:", supportedLang)
 // ...
 ```
 
-Possible errors:
+- Sending a Query does not add any events to the Workflow's Event History.
+- You can send a Query to a Workflow Execution that has closed.
```

Comment:
Should probably note that a Worker has to be running to handle it

Reply:
Thanks, that's included in the latest versions.